### PR TITLE
[IMPROVEMENT] Deprecate mod_deploy module and update Deployment pipeline workflow

### LINKED ISSUE #836: [PROPOSAL] Deprecate `/deploy` endpoint and related configuration
Please prefix your issue with one of the following: [BUG], [PROPOSAL], [QUESTION].

Sample platform commit (found at the bottom of each page) : **X.X.X**

**In raising this issue, I confirm the following (please check boxes, eg [X]):**

- [x] I have read and understood the [contributors guide](https://github.com/CCExtractor/sample-platform/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that the bug-fix I am reporting can be replicated, or that the feature I am suggesting isn't already present.
- [x] I have checked that the issue I'm posting isn't already reported.
- [x] I have checked that the issue I'm posting isn't already solved and no duplicates exist in [closed issues](https://github.com/CCExtractor/sample-platform/issues?q=is%3Aissue+is%3Aclosed) and in [opened issues](https://github.com/CCExtractor/sample-platform/issues)
- [x] I have checked the pull requests tab for existing solutions/implementations to my issue/suggestion.

**My familiarity with the project is as follows (check one, eg [X]):**

- [ ] I have never visited/used the platform.
- [ ] I have used the platform just a couple of times.
- [ ] I have used the platform extensively, but have not contributed previously.
- [x] I am an active contributor to the platform.

---

With #811, the platform would have automated deployments via GitHub workflows, therefore the `/deploy` endpoint needs to be deprecated.
The areas where changes need to be made are:
1. mod_deploy module
2. sample config
3. Installations scripts (related to `GITHUB_DEPLOY_KEY`)
4. Related tests
5. Github secrets

### PR DESCRIPTION
**Pull request by Tarun-Arora**
Please prefix your pull request with one of the following: **[FEATURE]** **[FIX]** **[IMPROVEMENT]**.

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [contributors guide](https://github.com/CCExtractor/sample-platform/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

**My familiarity with the project is as follows (check one):**

- [ ] I have never used the project.
- [ ] I have used the project briefly.
- [ ] I have used the project extensively, but have not contributed previously.
- [x] I am an active contributor to the project.

---

Closes #836 

Please remove the GitHub webhook for `/deploy` endpoint from the repository settings

### CODE REVIEW

File: decorators.py [hunk 1]
```diff
@@ -93,10 +94,9 @@ def decorated_function(*args, **kwargs):
             ctx['applicationName'] = 'CCExtractor CI platform'
             ctx['applicationVersion'] = getattr(g, 'version', 'Unknown')
             ctx['currentYear'] = date.today().strftime('%Y')
-            try:
-                from build_commit import build_commit
-            except ImportError:
-                build_commit = 'Unknown'
+            repo = git.Repo(search_parent_directories=True)
```
[the remark this answers is not in the document]
Comment (Tarun-Arora) [reply decorators.py:97]: Yes true, but this is also basically looking up of a file
We can put this with `g.github` globals, then we would have to lookup only once?

(Actually I wanted to remove the overhead of maintaining a file for build commit hash)

File: utility.py [hunk 2]
```diff
@@ -30,7 +36,110 @@ def serve_file_download(file_name, file_folder, file_sub_folder='') -> werkzeug.
...
+
+
+def request_from_github(abort_code: int = 418) -> Callable:
+    """Provide decorator to handle request from GitHub on the web hook."""
+    def decorator(f):
+        """Decorate the function to check if a request is a GitHub hook request."""
+        @wraps(f)
+        def decorated_function(*args, **kwargs):
+            if request.method != 'POST':
+                g.log.info(f"Request type {request.method} detected")
```
Comment (thealphadollar) [inline utility.py:52]: [nitpik] Make the logs lowercase.

File: utility.py [hunk 3]
```diff
@@ -30,7 +36,110 @@ def serve_file_download(file_name, file_folder, file_sub_folder='') -> werkzeug.
...
+    :param data: Signature's data
+    :type data: bytearray
+    :param private_key: Signature's token
+    :type private_key: str
+    """
+    hash_algorithm, github_signature = x_hub_signature.split('=', 1)
+    algorithm = hashlib.__dict__.get(hash_algorithm)
+    encoded_key = bytes(private_key, 'latin-1')
+    mac = hmac.new(encoded_key, msg=data, digestmod=algorithm)
+    return hmac.compare_digest(mac.hexdigest(), github_signature)
```
Comment (thealphadollar) [inline utility.py:145]:
```suggestion

    return hmac.compare_digest(mac.hexdigest(), github_signature)
```

File: utility.py [hunk 4]
```diff
@@ -30,7 +36,110 @@ def serve_file_download(file_name, file_folder, file_sub_folder='') -> werkzeug.
...
+
+    :param request_ip: The IP address the request came from.
+    :type request_ip: IPAddress
+    :return: True if the IP address is a valid GitHub Web Hook requester.
+    :rtype: bool
+    """
+    for block in get_cached_web_hook_blocks():
+        if request_ip in ip_network(block):
+            return True
+    return False
```
Comment (thealphadollar) [inline utility.py:104]:
```suggestion

    return False
```

File: utility.py [hunk 5]
```diff
@@ -30,7 +36,110 @@ def serve_file_download(file_name, file_folder, file_sub_folder='') -> werkzeug.
...
+                g.log.critical("User-Agent does not begin with GitHub-Hookshot/!")
+                abort(abort_code)
+
+            if not request.is_json:
+                g.log.critical("Request is not JSON!")
+                abort(abort_code)
+
+            return f(*args, **kwargs)
+        return decorated_function
+    return decorator
```
Comment (thealphadollar) [inline utility.py:75]:
```suggestion
            return f(*args, **kwargs)

        return decorated_function

    return decorator
```

**Review by thealphadollar — APPROVED** [review]
LGTM.

**Review by thealphadollar — CHANGES_REQUESTED** [review]
Let us add tests for the new changes and merge the PR afterwards.

**Review by thealphadollar — DISMISSED** [review]

**Review by thealphadollar — APPROVED** [review]

**Comment by thealphadollar** [comment]
@canihavesomecoffee Can you please review before we merge it?

**Review by canihavesomecoffee — APPROVED** [review]

**Pull request merged**